src/frontend/js/components/PaymentButton/index.spec.tsx

```diff
@@ -0,0 +1,472 @@
+import { act, fireEvent, render, screen, waitFor } from '@testing-library/react';
+import fetchMock from 'fetch-mock';
+import { PropsWithChildren } from 'react';
+import { IntlProvider } from 'react-intl';
+import { hydrate, QueryClient, QueryClientProvider } from 'react-query';
+import {
+  AddressFactory,
+  ContextFactory as mockContextFactory,
+  CreditCardFactory,
+  OrderWithOneClickPaymentFactory,
+  OrderWithPaymentFactory,
+  PersistedClientFactory,
+  ProductFactory,
+  QueryStateFactory,
+} from 'utils/test/factories';
+import JoanieApiProvider from 'data/JoanieApiProvider';
+import { CourseCodeProvider } from 'data/CourseCodeProvider';
+import { PAYMENT_SETTINGS } from 'settings';
+import type * as Joanie from 'types/Joanie';
+import { OrderState } from 'types/Joanie';
+import createQueryClient from 'utils/react-query/createQueryClient';
+import PaymentButton from '.';
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockContextFactory({
+    authentication: {
+      backend: 'fonzie',
+      endpoint: 'https://authentication.test',
+    },
+    joanie_backend: {
+      endpoint: 'https://joanie.test',
+    },
+  }).generate(),
+}));
+
+jest.mock('components/PaymentInterfaces');
+
+describe('PaymentButton', () => {
+  const formatPrice = (price: number, currency: string) =>
+    new Intl.NumberFormat('en', {
+      currency,
+      style: 'currency',
+    }).format(price);
+
+  const Wrapper = ({
+    client = new QueryClient(),
+    children,
+  }: PropsWithChildren<{ client?: QueryClient }>) => (
+    <IntlProvider locale="en">
+      <JoanieApiProvider>
+        <CourseCodeProvider code="00000">
+          <QueryClientProvider client={client}>{children}</QueryClientProvider>
+        </CourseCodeProvider>
+      </JoanieApiProvider>
+    </IntlProvider>
+  );
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    fetchMock.restore();
+    jest.clearAllTimers();
+    jest.resetAllMocks();
+  });
+
+  it('should render a payment button', () => {
+    const product: Joanie.Product = ProductFactory.generate();
+
+    render(
+      <Wrapper>
+        <PaymentButton product={product} onSuccess={jest.fn()} />
+      </Wrapper>,
+    );
+
+    const $button = screen.getByRole('button', {
+      name: `Pay ${formatPrice(product.price, product.price_currency)}`,
+    }) as HTMLButtonElement;
+
+    // As a billing address is missing, button should be disabled
+    expect($button.disabled).toBe(true);
+  });
+
+  it('should render a payment button with a specific label when a credit card is provided', () => {
```

Comment:
```suggestion
  it('should render a payment button with a specific label when a credit card is provided', () => {
```
you are testing two things the label and the fact the button is not enabled because the address is not provided.
For this test, I would like to be sure this specific button "Pay in one click" can be enabled, to make sure we can pay :-)

If you want to test the disabled/enabled maybe it could be dissociated

Reply:
I completed the test to check that button is enabled when an address is provided.